test/per_file_coverage.sh

```diff
@@ -63,7 +63,7 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/extensions/listener_managers/validation_listener_manager:70.5"
 "source/extensions/watchdog/profile_action:83.3"
 "source/server:91.0" # flaky: be careful adjusting. See https://github.com/envoyproxy/envoy/issues/15239
-"source/server/config_validation:89.2"
```

Comment:
I added `regexEngine()` to the `Server::Instance`. This isn't used (yet) in the config_validation path, because most of the uses of `StringMatcherImpl` still don't use it. Future PRs will convert all of those, and this line will get hit in coverage at that time.